aptos-move/framework/aptos-framework/sources/object_code_deployment.move

```diff
@@ -0,0 +1,73 @@
+module aptos_framework::object_code_deployment {
+    use std::bcs;
+    use std::features;
+    use std::signer;
+    use std::string;
+    use std::vector;
+    use aptos_framework::account;
+    use aptos_framework::code;
+    use aptos_framework::code::PackageRegistry;
+    use aptos_framework::object;
+    use aptos_framework::object::{ExtendRef, Object};
+
+    /// Object code deployment not supported.
+    const EOBJECT_CODE_DEPLOYMENT_NOT_SUPPORTED: u64 = 0x1;
+    /// Not the owner of the `PublisherRef`
+    const ENOT_OWNER: u64 = 0x2;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct PublisherRef has key {
+        extend_ref: ExtendRef,
+    }
+
```

Comment:
Add events as well

Reply:
Added in `object_code_deployment.move` and one in `code.move`